doc/src/explanation/glossary.md

```diff
@@ -0,0 +1,743 @@
+
+-->
+
+<!-- To cross reference a glossary item elsewhere in the documentation use
+{term}`termname` in Markdown or :term:`termname` in RST. Use
+{term}`Text <termname>` or :term:`text <termname>` to make a link with
+different text. -->
+
+```{glossary}
+
+Antiderivative
+
+    An *antiderivative* of a function $f(x)$ with respect to $x$ is an
+    function $F(x)$ such that $\frac{d}{dx}F(x) = f(x).$ It is also sometimes
+    called an "indefinite integral" of $f(x)$, and written as $\int
+    f(x)\,dx.$ Antiderivatives in SymPy can be computed with
+    {func}`~.integrate`. Any given function has an infinite family of
+    antiderivatives, typically obtained by adding an arbitrary constant, but
+    {func}`~.integrate` only returns one
+    such antiderivative, which is sometimes informally referred to as "the
+    antiderivative" of $f(x)$. Note some sources call this the "primitive" of
+    $f(x)$, but this terminology is not used in SymPy because it is not as
+    universally used as "antiderivative", and because "primitive" has other
+    meanings in mathematics.
+
+`args`
+
+    The *`args`* property of a SymPy {term}`expression` `expr` is a tuple of
+    the top-level {term}`subexpressions <subexpression>` used to create
+    `expr`. They are the arguments to the class used to create the expression.
+    The args of any expression can be obtained by the `.args` attribute. For
+    example, `(1 + x*y).args` is `(1, x*y)`, because it equals `Add(1, x*y)`.
+    The `args` together with {term}`func` completely define an expression. It
+    is always possible to walk the {term}`expression tree` and extract any
+    subexpression of a SymPy expression by repeated use of `.args`. Every
+    SymPy expression can be rebuilt exactly with `func` and `args`, that is,
+    `expr.func(*expr.args) == expr` will always be true of any SymPy
+    expression `expr`. The args of an expression may be the empty tuple `()`,
+    meaning the expression is an {term}`atom`.
+
+Assumptions
+
+    *Assumptions* are a set of predicates on a {term}`symbol` or
+    {term}`expression` that define the set of possible values it can take.
+    Some examples of assumptions are `positive`, `real`, and `integer`.
+    Assumptions are related to one another logically, for example, an
+    assumption of `integer` will always automatically imply `real`.
+    Assumptions use a {term}`three-valued logic` system where predicates are
+    either `True`, `False`, or `None`.
+
+    Assumptions are either *assumed* or *queried*. For example, a symbol `x`
+    might be *assumed* to be positive by defining it as `x = symbols('x',
+    positive=True)`. Then an assumption might be *queried* on the expression
+    containing this symbol, like `(x + 1).is_real`, which in this case would
+    return `True`.
+
+    If no assumptions are assumed on a symbol, then by default symbols are
+    assumed to be general complex numbers. Setting assumptions is important
```

Comment:
This statement is tricky because symbols are not automatically assumed to be finite.

Reply:
I do want the text here to be correct, but I also don't want to bog down the definitions with technical details that wouldn't matter to most people. 

The point here is just to be that nothing is assumed, not really about the "default assumptions". Maybe I can tweak the wording a bit to not imply that finite is assumed by default.